app/components/search/QueryResults.tsx

```diff
@@ -29,81 +29,74 @@ type QueryResultsProps<T> = {
    * The message to display when no results are found.
    */
   noResultsMessage?: string;
+  activeIndex?: number;
 };
 
-const QueryResults = forwardRef(
-  <T,>(
-    {
-      query,
-      matcher,
-      data,
-      onSelectResult = () => {},
-      keyExtractor = (datum) => datum?.toString() ?? "",
-      renderResult,
-      noResultsMessage = "No results",
-    }: QueryResultsProps<T>,
-    ref: Ref<HTMLUListElement>,
-  ) => {
-    const matcherForQuery = useCallback(matcher(query), [query]);
+function QueryResults<T>({
+  query,
+  matcher,
+  data,
+  onSelectResult = () => {},
+  keyExtractor = (datum) => datum?.toString() ?? "",
+  renderResult,
+  noResultsMessage = "No results",
+  activeIndex = 0,
+}: QueryResultsProps<T>) {
+  const matcherForQuery = useCallback(matcher(query), [query]);
 
-    const predicate = useCallback(
-      (datum: T) => {
-        return matcherForQuery(datum).matches;
-      },
-      [matcherForQuery],
-    );
+  const predicate = useCallback(
+    (datum: T) => {
+      return matcherForQuery(datum).matches;
+    },
+    [matcherForQuery],
+  );
 
-    const results = useMemo(() => {
-      return data.filter(predicate).sort((a, b) => {
-        const aScore = matcherForQuery(a).score;
-        const bScore = matcherForQuery(b).score;
+  const results = useMemo(() => {
+    return data.filter(predicate).sort((a, b) => {
+      const aScore = matcherForQuery(a).score;
+      const bScore = matcherForQuery(b).score;
 
-        // Higher is better
-        return bScore - aScore;
-      });
-    }, [data, predicate, matcherForQuery]);
+      // Higher is better
+      return bScore - aScore;
+    });
+  }, [data, predicate, matcherForQuery]);
 
-    return (
-      <div className="shadow-lg">
-        <ul ref={ref}>
-          {results.map((datum, index) => (
-            <li
-              key={keyExtractor(datum)}
-              onClick={() => onSelectResult(datum)}
-              className="bg-uclaBlue"
+  return (
+    <div className="shadow-lg">
+      <ul>
+        {results.map((datum, index) => (
+          <li
+            key={keyExtractor(datum)}
+            onClick={() => onSelectResult(datum)}
+            className="bg-uclaBlue"
+          >
+            <div
+              className={`hover:opacity-50 ${index === activeIndex ? "opacity-75" : ""}`}
```

Comment:
instead of manually constructing the classname string, let's use the [`classnames`](https://www.npmjs.com/package/classnames) package like in 

https://github.com/nathanhleung/uclagrades.com/blob/1ba1494f190fe5082bb4642d16e2177eafc575a1/app/components/Loading.tsx#L30-L32

falsy params will be omitted